### ISSUE TITLE
Bug: [prefer-nullish-coalescing] doesn't report on chain expressions in a ternary

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10531]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&esQuery=N4IgZglgNgLgpgJxALhCAvkA&code=JYOwLgpgTgZghgYwgAgEIHsAeyDeAoZZANzgBsBXCALmQGcwpQBzAbjwF888ATCBUuFBQxyIBGGDoQyJhDAAxMqQBGiANYZMACgCUNTWx58BQ5Ain1kveOVJhNNHMjiPky-VmQAfZKOugIbmR2YMNefkEUcxBLazhbeywAeQAHCSkyVzgAfldlXLRPHz8IGACgkPZDAHpq5CEU9CgwWjMmoXFSAE8jGztNADo4AeVkAEIAXmQQW1JkbKtS%2BP6sIZHkGlkFJVUEDSxdGrraAAt0WyCGprBe5cTMVPSQMiHs9cnp2fnFvvvHyWepFe602ckUpBU6k0hy4tToZwuzlItHQ9QgjWatwSmn%2BGSBOXeUxmEO%2BcWxyTSAJew1GoO2EN2%2B20OhYQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Box {
  value: string;
}

declare function getFallbackBox(): Box;

declare const defaultBox: { a: { b: Box | undefined } };

declare const defaultBoxOptional: { a?: { b?: Box | undefined } };

// reports correctly
defaultBox.a.b != null ? defaultBox.a.b : getFallbackBox();

// should report
defaultBoxOptional.a?.b != null ? defaultBoxOptional.a?.b : getFallbackBox();

// should also report
defaultBoxOptional.a?.b != null ? defaultBoxOptional.a.b : getFallbackBox();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-nullish-coalescing": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected all cases above to be reported with a suggested fix.

### Actual Result

Only the first case (member-expression) is reported.

### Additional Info

I think the third case (chain-expression with a member-expression as an alternate) is the more common case, as `no-unnecessary-condition` will flag the second case (chain-expression).

**Comment by kirkwaiblinger** [comment]
+1 to this. 

Noting that it's easy to analyze as long as we only think about the `!= null` case, but the implementation will have to pay close attention to the strict nullish equality cases. I guess `!== null` can never report in the presence of (non-unnecessary) intermediate optional member access and `!== undefined` can only if the LHS can never be `null`? Or something along those lines.

**Comment by OlivierZal** [comment]
I started working on https://github.com/typescript-eslint/typescript-eslint/pull/10517, I can address it right after in a new PR.

**Comment by OlivierZal** [comment]
Hi! I'll take this one once https://github.com/typescript-eslint/typescript-eslint/pull/10517 merged

**Comment by OlivierZal** [comment]
Quick analysis:
- `isNodeEqual` doesn't handle `ChainExpression`: https://github.com/typescript-eslint/typescript-eslint/pull/10708/files#diff-c5588b9e939ec001873faeffa7427caa2e1fecf0bafeffd5220964c9a4eedb41R32
- neither the rule: https://github.com/typescript-eslint/typescript-eslint/pull/10708/files#diff-dd40826685069133bdddab944ed6d6e71438ba0847484dd9a8ded0faba704283R23

Remaining work: make the rule determine when `a?.b` as `node.test` can be considered equal to `a.b` as `node.consequent` (or the equivalent with `!a?.b`).
